lldb-eval/ast.h

```diff
@@ -46,6 +46,12 @@ class AstNode {
   virtual bool is_literal_zero() const { return false; }
   virtual lldb::SBType result_type() const = 0;
 
+  // Is the node literal zero or `std::nullptr_t`, including their casted
+  // versions (e.g. `(int*)nullptr`)?
+  virtual bool is_literal_null() const {
+    return Type(result_type_deref()).IsNullPtrType();
```

Comment:
This will return `true` for non-literal as well. Do we want that? I think `return false` should be enough, since all literals are handles it `LiteralNode`.

Reply:
Actually the problem isn't with just literal `nullptr`, but any variable of type `std::nullptr_t` (and that can be many different node kinds, e.g. dereference: `*null_ptr_addr`). I could override the method for each of this node kinds separately, but I think this is a simple way to cover all the cases.

Hmm word "literal" is a bit misleading in the method name :/

Also, we only need this for cast nodes (we will never apply binary addition on nullptr or literal zero directly), so maybe we could have `is_casted_null` (or similar name) implemented only in `CStyleCastNode`? E.g. like this:
```cpp
bool CStyleCastNode::is_casted_null() const override {
  Type rhs_type = rhs_->result_type_deref();
  return rhs_type.IsNullPtrType() || rhs_->is_literal_zero() || rhs_->is_casted_null();
}
```